libraries/chain/block_log.cpp

```diff
@@ -834,6 +857,56 @@ namespace eosio { namespace chain {
       }));
    }
 
+   bool prune(pruned_transaction& ptx) { 
+      ptx.prune_all();
+      return true;
+   }
+   
+   void block_log::prune_transaction(uint32_t block_num, transaction_id_type id) {
+      try {
+         EOS_ASSERT(my->version >= 4, block_log_exception, "The block log version ${version} does not support transaction pruning.", ("version", my->version));
+         uint64_t pos = get_block_pos(block_num);
+         EOS_ASSERT(pos != npos, block_log_exception,
+                     "Specified block_num ${block_num} does not exist in block log.", ("block_num", block_num));
+
+         auto entry = my->read_v4_log_entry(pos);
+
+         EOS_ASSERT(entry.block->block_num() == block_num, block_log_exception,
+                     "Wrong block was read from block log.");
+
+         auto pruner = overloaded{[](transaction_id_type&) { return false; },
+                                  [&id](pruned_transaction& ptx) { return ptx.id() == id && prune(ptx); }};
+
+         for (auto trx : entry.block->transactions) {
+            if (trx.trx.visit(pruner)) {
+               my->block_file.seek(pos);
+               std::vector<char> buffer = detail::block_log_impl::pack(*entry.block, static_cast<pruned_transaction::cf_compression_type>(entry.compression.value));
+               my->block_file.write(buffer.data(), buffer.size());
+               my->block_file.flush();
+               return;
+            }
+         }
+
+         EOS_ASSERT(true, block_log_exception,
```

Comment:
I think you meant to write `false` here.  Also, don't use that pattern. Instead use `EOS_THROW`.

Reply:
Better yet return a boolean from this function rather than throwing.